File: Blind 75/Array/1. Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
  const previousNumbers = new Map();

  for (let i = 0; i < nums.length; i++) {
    const curNum = nums[i];
    if (previousNumbers.has(target - curNum)) {
      return [i, previousNumbers.get(target - curNum)];
    } else {
      previousNumbers.set(curNum, i);
    }
  }

  return [];
}

File: golang/random/1. Two Sum.go
package random

func twoSum(nums []int, target int) []int {
	previousNums := make(map[int]int)

	for firstIndex, num := range nums {
		diff := target - num

		secondIndex, ok := previousNums[diff]

		if ok {
			return []int{firstIndex, secondIndex}
		}

		previousNums[num] = firstIndex
	}

	return nil
}
